test/unit/org/apache/cassandra/locator/ReplicationFactorTest.java

```diff
@@ -52,32 +54,42 @@ private static void assertRfParseFailure(String s)
     private static void assertRfParse(String s, int expectedReplicas, int expectedTrans)
     {
         ReplicationFactor rf = ReplicationFactor.fromString(s);
-        Assert.assertEquals(expectedReplicas, rf.allReplicas);
-        Assert.assertEquals(expectedTrans, rf.transientReplicas());
-        Assert.assertEquals(expectedReplicas - expectedTrans, rf.fullReplicas);
+        assertEquals(expectedReplicas, rf.allReplicas);
+        assertEquals(expectedTrans, rf.transientReplicas());
+        assertEquals(expectedReplicas - expectedTrans, rf.fullReplicas);
     }
 
     @Test
-    public void parseTest()
+    public void shouldParseValidRF()
     {
+        assertRfParse("0", 0, 0);
         assertRfParse("3", 3, 0);
         assertRfParse("3/1", 3, 1);
-
         assertRfParse("5", 5, 0);
         assertRfParse("5/2", 5, 2);
+    }
 
+    @Test
+    public void shouldFailOnInvalidRF()
+    {
         assertRfParseFailure("-1");
         assertRfParseFailure("3/3");
+        assertRfParseFailure("3/-1");
         assertRfParseFailure("3/4");
+        assertRfParseFailure("3/");
```

Comment:
We could check the message of the exception caught in `assertRfParseFailure`, passing the expected message as an argument. Additionally, we could try strings producing `NumberFormatException`, like `""`, `"a/1"` and `"1/a"`:
```java
assertRfParseFailure("-1", "Replication factor must be non-negative, found -1");
assertRfParseFailure("3/3", "Transient replicas must be zero, or less than total replication factor. For 3/3");
assertRfParseFailure("3/-1", "Amount of transient nodes should be strictly positive, but was: '-1'");
assertRfParseFailure("3/4", "Transient replicas must be zero, or less than total replication factor. For 3/4");
assertRfParseFailure("3/", "Replication factor format is <replicas> or <replicas>/<transient>");
assertRfParseFailure("1/a", "For input string: \"a\"");
assertRfParseFailure("a/1", "For input string: \"a\"");
assertRfParseFailure("", "For input string: \"\"");
```
We could also consider throwing a `NumberFormatException` with a more RF-specific message, although that doesn't seem especially useful.